sqlglot/lineage.py

```diff
@@ -108,18 +109,8 @@ def to_node(
         if isinstance(scope.expression, exp.Union):
             upstream = upstream or Node(name="UNION", source=scope.expression, expression=select)
 
-            index = (
-                column
-                if isinstance(column, int)
-                else next(
-                    i
-                    for i, select in enumerate(scope.expression.selects)
-                    if select.alias_or_name == column
-                )
-            )
-
             for s in scope.union_scopes:
-                to_node(index, scope=s, upstream=upstream)
+                to_node(select.name, scope=s, upstream=upstream)
```

Comment:
These two don't seem equivalent - we have code that also handles the case where `column` is `int` above. What is the context behind this change?

Reply:
After my change the expressions that were setting index and select where the same except that index was returning an int or star and select a column or star expression, I did not see a reason to keep the duplicate code.